x-pack/plugins/observability_solution/infra/public/pages/link_to/use_node_details_redirect.ts

```diff
@@ -42,42 +45,68 @@ export const useNodeDetailsRedirect = () => {
   const {
     services: {
       application: { currentAppId$ },
+      share,
     },
   } = useKibanaContextForPlugin();
-
   const appId = useObservable(currentAppId$);
+  const locator = share.url.locators.get<AssetDetailsLocatorParams>(ASSET_DETAILS_LOCATOR_ID);
+
   const getNodeDetailUrl = useCallback(
     <T extends InventoryItemType>({
       assetType,
       assetId,
       search,
-    }: NodeDetailsRedirectParams<T>): LinkDescriptor => {
+    }: NodeDetailsRedirectParams<T>): RouterLinkProps => {
       const { from, to, ...additionalParams } = search;
+      const assetDetails =
+        Object.keys(additionalParams).length > 0
+          ? {
+              ...additionalParams,
+              dateRange: {
+                from: from ? new Date(parseFloat(from)).toISOString() : undefined,
+                to: to ? new Date(parseFloat(to)).toISOString() : undefined,
+              },
+            }
+          : {};
 
-      return {
-        app: 'metrics',
-        pathname: `link-to/${assetType}-detail/${assetId}`,
-        search: {
-          ...(Object.keys(additionalParams).length > 0
-            ? { [REDIRECT_ASSET_DETAILS_KEY]: rison.encodeUnknown(additionalParams) }
+      const _a = {
```

Comment:
What does `_a` mean?

Reply:
This is the name of the query param but good point about the naming - I renamed it to `queryParams` and put both parameters we use (`_a` and `assetDetails`) so we know what those values are.